### PR TITLE
fix transitive "colors" bad dependency

### PR DESCRIPTION
**Pull request by razbensimon**
### Issue & Steps to Reproduce / Feature Request
`colors` developer is crazy.
our cli could install the bad version of `colors` as a transitive dependency by `winston`.

### Solution
upgrade "winston" 3.3.4 - in that version, they locked to the previous and safe version of `colors`

### CODE REVIEW

**Comment by razbensimon** [comment]
[Our node CLI broke due to a corrupt unpinned version](https://trello.com/c/IRBgybce/5163-our-node-cli-broke-due-to-a-corrupt-unpinned-version)

**Review by RLRabinowitz — COMMENTED** [review]
@razbensimon Did you try `npm-shrinkwrap`? Not sure if it works with `yarn`, but we should use it. I see it's on your ticket as well

See [this](https://env0.slack.com/archives/CH9ND4Z6C/p1641802223157700?thread_ts=1641761667.138600&cid=CH9ND4Z6C)

**Comment by razbensimon** [comment]
@RLRabinowitz I saw it, and Im just in the missle of writing about it.
`shrinkwrap` is for npm users, but we already use yarn and have `yarn.lock` - which is 1:1 replacement (with better deterministic versions resolver FYI).
The thing here, is that Im not sure how we had a problem in Seed because our lock file is already locked to version 1.4.0 of `colors` - which is good.

**Comment by razbensimon** [comment]
@RLRabinowitz I talked with @yaronya and I mentioned that maybe it's because we used `npx` cli but I couldn't prove it yet.

**Comment by RLRabinowitz** [comment]
@razbensimon Where can I see this discussion?

**Comment by yaronya** [comment]
> @RLRabinowitz I talked with @yaronya and I mentioned that maybe it's because we used `npx` cli but I couldn't prove it yet.

Makes sense that `npx` ignored our `yarn.lock` so it didn't treat the locked versions under it (as according to the yarn docs, it should completely replaces the shrinkwrap file)

**Comment by yaronya** [comment]
@razbensimon since we're still using `npx` (and it's widely more used than using `yarn`'s alternative) to execute our CLI - I agree with @RLRabinowitz that we should consider the shrinkwrap file.

IMO, we can either:
1. Move from `yarn` to `npm` (`yarn.lock` -> `npm-shrinkwrap.json`)
2. Keep the `yarn.lock`, but make sure to create an aligned shrinkwrap file (aligned with `yarn.lock` I mean) by syncing them together using a `postinstall` script or something similar.

WDYT?

**Comment by razbensimon** [comment]
@yaronya see this: https://stackoverflow.com/questions/63266392/does-npx-use-yarn
it should work with `yarn insatll` IIUC

**Comment by yaronya** [comment]
> Yarn team didn't provide a npx equivalent deliberately

@razbensimon isn't `yarn dlx` is exactly it?

> @yaronya see this: stackoverflow.com/questions/63266392/does-npx-use-yarn
it should work with yarn insatll IIUC

❓ 😕

**Comment by razbensimon** [comment]
- TL/DR - We now use `shrinkwrap` file which will lock the npm module.
- We tested it with 3 mock new npm libs which dependent on each other[X -> Y(env0-cli) -> Z (colors)]
- We actually saw that npm publish ignores `yarn.lock` and won’t push it to npm.
- MEANING `env0/cli` never locked anything - and that was the actual problem.
- Even when we tell npm to include `yarn.lock` in the package, it won’t consider by `npx` 🤷🏼‍♂️ 
- `shrinkwrap` file is automatically pushed to npm.
- `yarn` doesn’t have an equivalent
- point to think about: we didn’t see big CLIs/packages that use shrinkwrap. so we have a feeling that we miss something  about the yarn ecosystem here - but not important for now and not worth our time

**Review by avnerenv0 — APPROVED** [review]
I didn't really dive into the whole `shrinkwrap` conversation above here, but LGTM
